pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -287,8 +287,7 @@ public void run() {
               new RealtimeSegmentConverter(_realtimeSegment, tempSegmentFolder.getAbsolutePath(), schema,
                   _tableNameWithType, tableConfig, realtimeSegmentZKMetadata.getSegmentName(), _sortedColumn,
                   HLRealtimeSegmentDataManager.this._invertedIndexColumns, _noDictionaryColumns,
-                  _varLengthDictionaryColumns, null/*StarTreeIndexSpec*/,
-                  indexingConfig.isNullHandlingEnabled()); // Star tree not supported for HLC.
+                  _varLengthDictionaryColumns, indexingConfig.isNullHandlingEnabled());
```

Comment:
We should support text index for HLC

Reply:
It is there for LLC already. Will add for HLC as well